cpp/include/cudf/detail/utilities/column.hpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/table/table_view.hpp>
+
+namespace cudf::detail {
+
+struct linked_column_view;
+
+using LinkedColPtr    = std::shared_ptr<linked_column_view>;
+using LinkedColVector = std::vector<LinkedColPtr>;
+
+/**
+ * @brief column_view with the added member pointer to the parent of this column.
+ *
+ */
+struct linked_column_view : public column_view {
+  // TODO(cp): we are currently keeping all column_view children info multiple times - once for each
+  //       copy of this object. Options:
+  // 1. Inherit from column_view_base. Only lose out on children vector. That is not needed.
+  // 2. Don't inherit at all. make linked_column_view keep a reference wrapper to its column_view
+  linked_column_view(column_view const& col) : linked_column_view(nullptr, col) {}
+
+  linked_column_view(linked_column_view* parent, column_view const& col)
+    : column_view(col), parent(parent)
+  {
+    children.reserve(col.num_children());
+    std::transform(
+      col.child_begin(), col.child_end(), std::back_inserter(children), [&](column_view const& c) {
+        return std::make_shared<linked_column_view>(this, c);
+      });
+  }
+
+  linked_column_view* parent;  //!< Pointer to parent of this column. Nullptr if root
+  LinkedColVector children;
```

Comment:
Our "view" types are already sketchy by having a `std::vector` member, but having a `std::shared_ptr` member makes it extra sketchy. This is no longer anything close to a cheap to copy object.

I'll need to read on to see why this is even necessary.

Reply:
> This is no longer anything close to a cheap to copy object.

It's not meant to be copied. It's meant to be accessed via its shared_ptr. When you copy a `column_view` the children vector needs to be duplicated. And those may contain children of their own. When you pass around a `LinkedColPtr`, none of that duplication happens.

Reply:
Now, after https://github.com/rapidsai/cudf/pull/10289/commits/6a282a61f25b4f8e9758731d8d2639757eb85ce1, `linked_column_view` inherits from `column_view_base` and so it's actually cheaper to handle than `column_view`.

Reply:
How about I delete the copy ctor and assignment operator just for good measure.